File: backend/src/main/java/com/github/mahambach/backend/model/MapMarkerDto.java
package com.github.mahambach.backend.model;

import com.github.mahambach.backend.model.enums.MapMarkerStatus;
import com.github.mahambach.backend.model.enums.Visibility;

public record MapMarkerDto(
        String worldMapId,
        String name,
        int xPosition,
        int yPosition,
        String markerTypeId,
        String playerArticleId,
        String gmArticleId,
        MapMarkerStatus status,
        Visibility visibility
) {
}

File: frontend/src/types/MapMarkerDto.ts
import {Visibility} from "../data/Visibility.ts";
import {MapMarkerStatus} from "../data/MapMarkerStatus.ts";

export type MapMarkerDto = {
    worldMapId: string,
    name: string,
    xPosition: number,
    yPosition: number,
    markerTypeId: string,
    status: MapMarkerStatus,
    visibility: Visibility
};

export const emptyMapMarkerDto:MapMarkerDto = {
    worldMapId: '',
    name: '',
    xPosition: 0,
    yPosition: 0,
    markerTypeId: '',
    status: 'ACTIVE',
    visibility: 'OWNER_ONLY'
};
